arch/arm/arm64/head.S

```diff
@@ -85,6 +104,170 @@ ENTRY(_start)
     load_paddr  x0, __start_bss
     bl      flush_dcache_range
 
-    /* Start an infinite loop */
-    PRINT("- Infinite loop -\n")
-2:  b   2b
+    PRINT("- Setup stack -\n")
+    ldr     x1, =stack_end
+    mov     sp, x1
+
+    /* Save boot arguments */
+    ldr     x0, =boot_data
+    stp     x21, x20, [x0]
+
+    PRINT("- Jump to C world-\n")
+    b       xtf_main
+ENDFUNC(_start)
+
+ENTRY(_mmu_disable)
+    dsb     sy
+
+    /* Turn off D-cache and MMU */
+    mrs     x2, sctlr_el1
+    bic     x2, x2, #SCTLR_M
```

Comment:
The Image protocol mandates that the MMU is off before jumping to the kernel. So I think you can drop this line.

Reply:
ok